cni/pkg/ambient/net.go

```diff
@@ -296,3 +298,36 @@ func (s *Server) DelPodFromMesh(pod *corev1.Pod) {
 func SetProc(path string, value string) error {
 	return os.WriteFile(path, []byte(value), 0o644)
 }
+
+func (s *Server) cleanStaleIPs(stales sets.Set[string]) {
+	log.Infof("Ambient stale Pod IPs to be cleaned: %s", stales)
+	switch s.redirectMode {
+	case IptablesMode:
```

Comment:
why is there no cleanup for iptables mode? don't we need to remove the pod from IPset?

Reply:
will add the removal to make logic consistently